jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/Response.java

```diff
@@ -317,9 +341,22 @@ static void sendRedirect(Request request, Response response, Callback callback,
             }
         }
 
-        response.getHeaders().put(HttpHeader.LOCATION, toRedirectURI(request, location));
+        if (content == null && request.getConnectionMetaData().getHttpConfiguration().isGenerateRedirectBody())
```

Comment:
If `isGenerateRedirectBody()` is false, then the trailers for this response should also not be produced.

Reply:
We can still to trailers.  This is a different "no body" that for responses that have no body. Rather it is saying "no body" as in we do not generate a HTML redirection message (as we never have done), but it is still a message that can have a body, so trailers can still be sent.    The "no body at all" handling is done in response generation